app/src/main/java/com/github/sdpcoachme/CoachesListActivity.kt

```diff
@@ -43,77 +42,66 @@ class CoachesListActivity : ComponentActivity() {
         super.onCreate(savedInstanceState)
 
         val isViewingContacts = intent.getBooleanExtra("isViewingContacts", false)
-        // TODO: update this to be current device location
-        val currentLat = 46.519054480712015
-        val currentLong = 6.566757578464391
         val database = (application as CoachMeApplication).database
+        val userLatLng = (application as CoachMeApplication).userLocation.value?: CAMPUS
+
         val futureListOfCoaches =
             if (isViewingContacts) {
                 database.getChatContacts(email = database.getCurrentEmail())
             } else {
                 database
                 .getAllUsersByNearest(
-                    latitude = currentLat,
-                    longitude = currentLong
+                    latitude = userLatLng.latitude,
+                    longitude = userLatLng.longitude
                 ).thenApply {
                     it.filter { user -> user.coach }
                 }
             }
 
-        setContent {
-            var listOfCoaches by remember { mutableStateOf(listOf<UserInfo>()) }
+        val email = database.getCurrentEmail()
 
-            // Proper way to handle result of a future in a Composable.
-            // This makes sure the listOfCoaches state is updated only ONCE, when the future is complete
-            // This is because the code in LaunchedEffect(true) will only be executed once, when the
-            // Composable is first created (given that the parameter key1 never changes). The code won't
-            // be executed on every recomposition.
-            // See https://developer.android.com/jetpack/compose/side-effects#rememberupdatedstate
-            LaunchedEffect(true) {
-                listOfCoaches = futureListOfCoaches.await()
+        if (email.isEmpty()) {
```

Comment:
I think it would make more sense to check if email is empty before using it at line 50 in futureListOfCoaches. As we check it, better do it as early as possible. (even though an empty email given to getChatContacts will trigger an IllegalArgumentException)